tst/EnergyPlus/unit/OutputProcessor.unit.cc

```diff
@@ -5901,6 +6068,27 @@ namespace OutputProcessor {
                                    "\"Output:Meter\". Will report to both bar.eso and bar.mtr"});
         compare_err_stream(errMsg);
     }
+
+    TEST_F(EnergyPlusFixture, DataOutputs_isKeyRegexLike)
+    {
+        std::vector<std::pair<std::string, bool>> test_cases{
+            {"*", false},
+            {"This is the first one", false},
+            {"This is another.*one", true},
+            {"This is (a|some) ones?", true},
+            {"Zone 1.1", true},   // The `.` could mean any character (though in this case it's not meant as a regex really)
+            {"Cafétéria", false}, // !std::isalnum, but not a regex
+        };
+
+        for (auto &[s, expectedIsRegexLike] : test_cases) {
+            EXPECT_EQ(expectedIsRegexLike, DataOutputs::isKeyRegexLike(s)) << "isKeyRegexLike: Failed for " << s;
+        }
+
+        for (auto &[s, expectedIsRegexLike] : test_cases) {
+            EXPECT_EQ(expectedIsRegexLike, DataOutputs::isKeyRegexLikeOri(s)) << "isKeyRegexLikeOri: Failed for " << s;
+        }
+    }
```

Comment:
The original implementation fails for `Cafétéria` (assumes it's a regex when there's no chance it is). isKeyRegexLikeOri is extracted from https://github.com/NREL/EnergyPlus/blob/2a26c7b350e59c5500b1d33dcebcf032815566a6/src/EnergyPlus/DataOutputs.cc#L141-L146)

Reply:
Test bed: https://godbolt.org/z/Y4n74orbK

Reply:
Let me ask a high-level question. To what extent do EnergyPlus users actually use general-purpose regex for variable keys? Or do they mostly use just plain `*`?

Reply:
Can you define "general-purpose" please? Not sure I follow.

Reply:
I guess I meant "general" rather than "general-purpose". Basically, anything other than a simple "*"  with nothing else around it. If those are the majority of the regex use cases, then you can probably create a special case for it that doesn't use regex at all.

Reply:
so you mean `.*` or `.+`? I doubt people use that, they just use `*` then.

I'd bet most use case  are prefix/suffix filtering. Eg if you name your zones `FLOOR1-Zone1` etc, you can get repporting for `FLOOR1-.*`. Same if you named the nodes of a loop like `LOOPNAME NODENAME`

Reply:
A prefix is also a special case that doesn't need the regex library.

Reply:
I don't know if you know what I am getting at. If the cost is in compiling the regex and the common usages are special cases that can be handled using `has_prefix`, `has_suffix`, or even no explicit matching at all (`*`), then we can reduce the frequency of regex compilation even further.  I'm going to just leave this here. Maybe what you've already done (:fire:) is the 80/20 sweet spot here and no additional optimization is needed at this time.